spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/TextEncryptorLocator.java

```diff
@@ -0,0 +1,25 @@
+package org.springframework.cloud.config.server.encryption;
+
+import org.springframework.security.crypto.encrypt.TextEncryptor;
+
+public class TextEncryptorLocator {
+
+    private TextEncryptor encryptor;
+
+    public TextEncryptorLocator() {
+    }
+
+    public TextEncryptorLocator(TextEncryptor encryptor) {
+        this.encryptor = encryptor;
+    }
+
+    // temporary solution to support EncryptionController REST API
+    public void setEncryptor(TextEncryptor encryptor) {
+        this.encryptor = encryptor;
+    }
+
+    public TextEncryptor get() {
```

Comment:
Not sure I see the benefit of this abstraction yet.  Can you explain a bit what you're thinking here?